src/OrchardCore.Modules/OrchardCore.Media/Assets/js/MediaField/attachedMediaField.js

```diff
@@ -129,68 +129,83 @@ function initializeAttachedMediaField(el, idOfUploadButton, uploadAction, mediaI
-                            data.result.files[i].isNew = true;
-                            //if error is defined probably the file type is not allowed
-                            if (data.result.files[i].error === undefined || data.result.files[i].error === null) {
-                                data.result.files[i].attachedFileName = data.files[i].attachedFileName;
-                                newMediaItems.push(data.result.files[i]);
+                        if (data.result.files.length > 0) {
+                            for (var i = 0; i < data.result.files.length; i++) {
+                                data.result.files[i].isNew = true;
+                                //if error is defined probably the file type is not allowed
+                                if (data.result.files[i].error === undefined || data.result.files[i].error === null) {
+                                    data.result.files[i].attachedFileName = data.files[i].attachedFileName;
+                                    newMediaItems.push(data.result.files[i]);
+                                }
+                                else
+                                    errormsg += data.result.files[i].error + "\n";
                             }
-                            else
-                                errormsg += data.result.files[i].error + "\n";
                         }
-                    }
                     
-                    if (errormsg !== "") {
-                        alert(errormsg);
-                        return;
-                    }
-                    console.log(newMediaItems);
-                    if (newMediaItems.length > 1 && allowMultiple === false) {
-                        alert($('#onlyOneItemMessage').val());
-                        mediaFieldApp.mediaItems.push(newMediaItems[0]);
-                        mediaFieldApp.initialized = true;
-                    } else {
-                        mediaFieldApp.mediaItems = mediaFieldApp.mediaItems.concat(newMediaItems);
-                        mediaFieldApp.initialized = true;
+                        if (errormsg !== "") {
+                            alert(errormsg);
+                            return;
+                        }
+                        console.log(newMediaItems);
+                        if (newMediaItems.length > 1 && allowMultiple === false) {
+                            alert($('#onlyOneItemMessage').val());
+                            mediaFieldApp.mediaItems.push(newMediaItems[0]);
+                            mediaFieldApp.initialized = true;
+                        } else {
+                            mediaFieldApp.mediaItems = mediaFieldApp.mediaItems.concat(newMediaItems);
+                            mediaFieldApp.initialized = true;
+                        }
+                    },
+                    error: function (jqXHR, textStatus, errorThrown) {
+                        console.log('error on upload!!');
```

Comment:
I know you didn't write this but:

```suggestion
                        console.log('Error on upload.');
```